File: baekjoon/implement/11723.py
"""
1. 문제유형
- 백준 11723 : 집합
- https://www.acmicpc.net/problem/11723

2. 알고리즘
- 설명에 따라 메서드를 추가한다.
3. 실패
- 비트연산으로 풀어야한다. 시간초과
"""

import sys

num = int(input())
s = set()
for _ in range(num):
  com = list(input().split())
  if len(com) == 1:
    if com[0] == 'all':
      s = set([i for i in range(1, 21)])
    elif com[0] == 'empty':
      s.clear
  else :
    if com[0] == 'add':
      s.add(int(com[1]))
    elif com[0] == 'remove':
      s.discard(int(com[1]))
    elif com[0] == 'check':
      if int(com[1]) in s:
        print(1)
      else:
        print(0)
    elif com[0] == 'toggle':
      if int(com[1]) in s:
        s.remove(int(com[1]))
      else:
        s.add(int(com[1]))
print(s)

File: baekjoon/implement/11723.java
import java.util.*;
import java.io.*;

class Main {
  
  public static void main(String[] args) throws IOException {

    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    StringTokenizer st;
    int c = Integer.parseInt(br.readLine());

    HashSet<Integer> hs = new HashSet<>();

    for(int i=0; i<c; i++) {
      st = new StringTokenizer(br.readLine());
      String op = st.nextToken();
      if("all".equals(op)) {
        hs = new HashSet<>(Arrays.asList(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20));
      } else if("empty".equals(op)) {
        hs.clear();
      } else {
        int x = Integer.parseInt(st.nextToken());
        if("add".equals(op)) {
          hs.add(x);
        } else if("remove".equals(op)) {
          hs.remove(x);
        } else if("check".equals(op)) {
          if(hs.contains(x)) {
            bw.write("1\n");
          } else {
            bw.write("0\n");
          }
        } else if("toggle".equals(op)) {
          if(hs.contains(x)) {
            hs.remove(x);
          } else {
            hs.add(x);
          }
        }
      }
    }
    bw.flush();
    bw.close();
    
  }
}
